nifi-nar-bundles/nifi-scripting-bundle/nifi-scripting-processors/src/main/java/org/apache/nifi/processors/script/ExecuteScript.java

```diff
@@ -0,0 +1,585 @@
+                .build();
+    }
+
+    /**
+     * Determines whether the given path refers to a valid file
+     *
+     * @param path a path to a file
+     * @return true if the path refers to a valid file, false otherwise
+     */
+    private boolean isFile(final String path) {
+        return path != null && Files.isRegularFile(Paths.get(path));
+    }
+
+    /**
+     * Performs setup operations when the processor is scheduled to run. This includes evaluating the processor's
+     * properties, as well as reloading the script (from file or the "Script Body" property)
+     *
+     * @param context the context in which to perform the setup operations
+     */
+    @OnScheduled
+    public void setup(final ProcessContext context) {
+        scriptEngineName = context.getProperty(SCRIPT_ENGINE).getValue();
+        scriptPath = context.getProperty(SCRIPT_FILE).getValue();
+        scriptBody = context.getProperty(SCRIPT_BODY).getValue();
+        modulePath = context.getProperty(MODULES).getValue();
+        setupEngine();
+    }
+
+    /**
+     * Configures the specified script engine. First, the engine is loaded and instantiated using the JSR-223
+     * javax.script APIs. Then, if any script configurators have been defined for this engine, their init() method is
+     * called, and the configurator is saved for future calls.
+     *
+     * @see org.apache.nifi.processors.script.ScriptEngineConfigurator
+     */
+    private void setupEngine() {
+        ClassLoader originalContextClassLoader = Thread.currentThread().getContextClassLoader();
+        try {
+            ProcessorLog log = getLogger();
+
+            // Need the right classloader when the engine is created. This ensures the NAR's execution class loader
+            // (plus the module path) becomes the parent for the script engine
+            ClassLoader scriptEngineModuleClassLoader = createScriptEngineModuleClassLoader(modulePath);
+            if (scriptEngineModuleClassLoader != null) {
+                Thread.currentThread().setContextClassLoader(scriptEngineModuleClassLoader);
+            }
+            scriptEngine = getScriptEngine();
+            ServiceLoader<ScriptEngineConfigurator> configuratorServiceLoader =
+                    ServiceLoader.load(ScriptEngineConfigurator.class);
+            for (ScriptEngineConfigurator configurator : configuratorServiceLoader) {
+                String configuratorScriptEngineName = configurator.getScriptEngineName();
+                try {
+                    if (configuratorScriptEngineName != null
+                            && configuratorScriptEngineName.equals(scriptEngineName)) {
+                        configurator.init(scriptEngine, modulePath);
+                        scriptEngineConfiguratorMap.put(configurator.getScriptEngineName(), configurator);
+                    }
+                } catch (ScriptException se) {
+                    log.error("Error initializing script engine configurator {}",
```

Comment:
I recommend just passing the Exception to the logger here and removing the if (log.isDebugEnabled()) ... the ProcessorLog itself will handle this, so that the Exception is logged only when debug is enabled.